### PR TITLE
Dynamic write-batch size and new serializer

### PR DESCRIPTION
**Pull request by lbradstreet**

### CODE REVIEW

File: src/onyx/messaging/aeron/subscriber.clj [hunk 1]
```diff
@@ -75,6 +82,7 @@
     (let [error-handler (reify ErrorHandler
                           (onError [this x] 
                             (reset! error x)
+                            (info "error" x)
```
Comment (MichaelDrogalis) [inline src/onyx/messaging/aeron/subscriber.clj]: Switch to an error Timbre call.

File: src/onyx/plugin/messaging_output.clj [hunk 2]
```diff
@@ -61,9 +27,37 @@
...
+        enc (pub/segment-encoder pub)]
+    (if (zero? (segment-encoder/segment-count enc))
+      (.position ^io.aeron.Publication (.publication pub))
+      (let [segments-len (- (segment-encoder/offset enc)
+                            (base-encoder/length base-enc))] 
+        (base-encoder/set-payload-length base-enc segments-len)
+        (pub/offer! pub (.buffer base-enc) (segment-encoder/offset enc) epoch)))))
+
+(defn try-sends [pubs epoch]
+  (loop [pubs pubs]
```
Comment (MichaelDrogalis) [inline src/onyx/plugin/messaging_output.clj]: Is this a possible hanging point if we encounter an unclean shutdown?

File: src/onyx/messaging/aeron/subscriber.clj (hunk 1 above)
Comment (lbradstreet) [inline src/onyx/messaging/aeron/subscriber.clj]: This line should be removed.

File: src/onyx/plugin/messaging_output.clj (hunk 2 above)
Comment (lbradstreet) [inline src/onyx/plugin/messaging_output.clj]: Nope, it'll stop recurring if there is a negative return code.

**Pull request merged**